### DISCUSSION TITLE
Primary Key is not editable even if readonly = false

### DISCUSSION THREAD

**Discussion by powerseed** [directus/directus#10354] [Archive]
I found that a table's PK cannot be altered even if its `readonly` attribute is `false`.

The screenshot below is the column, which is inactive.

![Screenshot_2](https://user-images.githubusercontent.com/42278237/144959704-2602b190-b9a5-496b-853a-aca4f9d706b3.png)

(This column is also a foreign key if this is relavant.)

And in its config, the `readonly` field is not checked.

![Screenshot_3](https://user-images.githubusercontent.com/42278237/144959710-219ecea5-fbda-482d-9bfd-c8edcbc3f0e2.png)

I am wondering if this is a bug or I configured incorrectly somewhere.

Thanks!

**Comment by joselcvarela** [comment]
Hello @powerseed 
I believe this is way it is to prevent failures when used with foreign keys.
If you change a primary key that is referenced on another table, the database will just throw an error complaining about the field is still referenced or something alike.

I also found out changing primary key useful on the past, but now I think the best approach is to always try to use an auto primary key (auto-incremented or randomly generated).

Maybe that readonly field should also be locked to true on case of primary keys.

**Reply by rijkvanzanten** [reply]
This is correct. If you've configured the collection with auto increment or UUID, the primary key field will be force-locked. If you're using a manually generated string for the primary key, the input works like any other.
